docs/admin/reference-architectures.md

```diff
@@ -0,0 +1,163 @@
+infrastructure resources offered by cloud providers.
+
+### Proxy
+
+A workspace proxy serves as a relay connection option for developers connecting
+to their workspace over SSH, a workspace app, or through port forwarding. It
+helps reduce network latency for geo-distributed teams by minimizing the
+distance network traffic needs to travel. Notably, workspace proxies do not
+handle dashboard connections or API calls.
+
+### Provisioner
+
+Provisioners in Coder execute Terraform during workspace and template builds.
+While the platform includes built-in provisioner daemons by default, there are
+advantages to employing external provisioners. These external daemons provide
+secure build environments, and reduce server load, improving performance and
+scalability. Each provisioner can handle a single concurrent workspace build,
+allowing for efficient resource allocation and workload management.
+
+### Registry
+
+The Coder Registry hosts starter templates for various cloud providers and
+orchestration platforms, enabling self-service cloud development environments
+via Terraform-defined infrastructure. Additionally, Coder introduces _Modules_
+to streamline template creation by extracting commonly used functionalities such
+as web IDEs, third-party integrations, and helper scripts into reusable
+components.
+
+The Registry is hosted service and it is not available for air-gapped
+deployments.
+
+## Scale tests methodology
+
+Scaling Coder involves careful planning and testing to ensure it can handle more
+load without compromising service. This process encompasses infrastructure
+setup, traffic projections, and aggressive testing to identify and mitigate
+potential bottlenecks.
+
+A dedicated cluster for Coder is Kubernetes cluster specifically configured to
+host and manage Coder workloads. Kubernetes provides container orchestration
+capabilities, allowing Coder to efficiently deploy, scale, and manage workspaces
+across a distributed infrastructure. This ensures high availability, fault
+tolerance, and scalability for Coder deployments. The cluster can be deployed
+using the Helm chart.
+
+Our scale tests include the following stages:
+
+1. Prepare environment: create expected users and provision workspaces.
+
+2. Dashboard evaluation: verify the responsiveness and stability of Coder
+   dashboards under varying load conditions. This is achieved by simulating user
+   interactions using instances of headless Chromium browsers.
+
+3. SSH connections: establish user connections with agents, verifying their
+   ability to echo back received content.
+
+4. Workspace application traffic: assess the handling of user connections with
+   specific workspace apps, confirming their capability to echo back received
+   content effectively.
+
+5. Cleanup: delete workspaces and users created in step 1.
+
+### Infrastructure and setup requirements
+
+In a single workflow, the scale tests runner evenly spreads out the workload
+like this:
```

Comment:
@johnstcn Thank you for proofreading! I'm curious if we need to explain this scale tests model a bit. Scale tests are performed in stages (SSH, Web Terminal, apps), but here we claim that this is continuous traffic. WDYT?

Reply:
This is probably a good follow-up PR.

Also @mafredri correct me if I'm wrong, but it is both continuous and in stages, correct?

```
     | T0 | T1 | T2 | T3 |
SSH  |  X |  X |  X |  X |
PTY  |    |  X |  X |  X |
APP  |    |    |  X |  X |
CHRO |    |    |    |  X |
```

Also, this is a configurable parameter on the template.

Reply:
Yeah, that's right @johnstcn, although I'd expand the table:

```
     | T0 | T1 | T2 | T3 | T4 | T5 | T6 |
SSH  |  X |  X |  X |  X |    |    |    |
PTY  |    |  X |  X |  X |  X |    |    |
APP  |    |    |  X |  X |  X |  X |    |
CHRO |    |    |    |  X |  X |  X |  X |
```

Reply:
Thanks, rephrased it and added the table 👍 !